clone/cmd/sumdbclone/sumdbclone.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sumdbclone is a one-shot tool for downloading entries from sum.golang.org.
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/clone/internal/cloner"
+	"github.com/google/trillian-examples/clone/internal/verify"
+	"github.com/google/trillian-examples/clone/logdb"
+	sdbclient "github.com/google/trillian-examples/sumdbaudit/client"
+	"github.com/google/trillian/merkle/rfc6962"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+var (
+	height         = flag.Int("h", 8, "tile height")
+	vkey           = flag.String("k", "sum.golang.org+033de0ae+Ac4zctda0e5eza+HJyk9SxEdh+s3Ux18htTTAD8OuAn8", "key")
+	mysqlURI       = flag.String("mysql_uri", "", "URL of a MySQL database to clone the log into. The DB should contain only one log.")
+	writeBatchSize = flag.Uint("write_batch_size", 1024, "The number of leaves to write in each DB transaction.")
+	workers        = flag.Uint("workers", 4, "The number of worker threads to run in parallel to fetch entries.")
+)
+
+func main() {
+	flag.Parse()
+
+	if len(*mysqlURI) == 0 {
+		glog.Exit("Missing required parameter 'mysql_uri'")
+	}
+
+	ctx := context.Background()
```

Comment:
nitty: could move this down to L63?

Reply:
I prefer to determine the context up near the top of main functions. It's possible that the other init stuff would take it in the future, and then it's just going to move up. I don't think moving it down adds any clarity.